test/ranges.jl

```diff
@@ -1,26 +1,109 @@
-    @test r === range(                stop=last(r), length=length(r))
-    @test r === range(first(r),       last(r)                       )
-    # the next ones use ==, because it changes the eltype
-    @test r ==  range(first(r),       last(r),      length(r)       )
-    @test r ==  range(start=first(r), stop=last(r), length=length(r))
+        r = 1:5
+        let start=first(r), step=step(r), stop=last(r), length=length(r)
+            @test r === range(; start, stop              )
+            @test r === range(;        stop,       length)
+            # the next three lines uses ==, because it changes the eltype
+            @test r ==  range(; start, stop,       length)
+            @test r ==  range(; start,       step, length)
+        end
+        for T = (Int8, Rational{Int16}, UInt32, Float64, Char)
+            @test typeof(range(start=T(5), length=3)) === typeof(range(stop=T(5), length=3))
+        end
+    end
+    @testset "range(start, stop)" begin
+        r = 2:11
+        let start=first(r), stop=last(r), step=step(r), length=length(r)
+            q = range(start, stop)
+            @test q === r
+            @test eltype( range( UInt8(start),  UInt8(stop) ) ) == UInt8
+            @test eltype( range( UInt8(start), BigInt(stop) ) ) == BigInt
+            @test eltype( range( UInt8(start),    Int(stop) ) ) == Int
+            # One positional argument
+            @test q === range(  start; stop             )
+            @test q === range(  start;            length)
+            # Two positional arguments
+            @test q === range(  start, stop             )
+            @test q ==  range(  start, stop; step       )
+            @test q ==  range(  start, stop;      length)
+            # All keyword arguments
+            @test q === range(; start, stop             )
+            @test q ==  range(; start, stop, step       )
+            @test q ==  range(; start, stop,      length)
+            @test q ==  range(;        stop,      length)
+        end
 
-    for T = (Int8, Rational{Int16}, UInt32, Float64, Char)
-        @test typeof(range(start=T(5), length=3)) === typeof(range(stop=T(5), length=3))
+        r = 2.8:20.0
+        let start=first(r), stop=last(r), step=step(r), length=length(r)
+            q = range(start, stop)
+            @test q === r
+            # One positional argument
+            @test q === range(  start; stop             )
+            @test q === range(  start;            length)
+            # Two positional arguments
+            @test q === range(  start, stop             )
+            @test q ==  range(  start, stop; step       )
+            @test q ==  range(  start, stop;      length)
+            # All keyword arguments
+            @test q === range(; start, stop             )
+            @test q ==  range(; start, stop, step       )
+            @test q ==  range(; start, stop,      length)
+            @test q ≈   range(;        stop,      length)
```

Comment:
Should we add `@test_broken q ==   range(;        stop,      length)`? Not sure, equality seems very hard to achieve here.

Reply:
Actually on master I currently get equality:
```julia
julia> r = 2.8:20.0
2.8:1.0:19.8

julia> using Test

julia> let start=first(r), stop=last(r), step=step(r), length=length(r)
           q = range(start, stop)
           @test q == range(; stop, length)
       end
Test Passed

julia> versioninfo()
Julia Version 1.7.0-DEV.551
Commit 9daf90ab49* (2021-02-16 19:40 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)

```